### DISCUSSION TITLE
CH4, CO2, brine

### DISCUSSION THREAD

**Discussion by mortezaaesmaeilpour** [idaholab/moose#19409] [Q&A Modules: General] [answered]
Dear MOOSE team

I'm going to use the PorousFlow module to simulate multiphase flow in a reservoir. The main idea is to inject a mixture of brine and CO2 into a reservoir. the reservoir is initially filled with brine and CH4. since the solubility of CO2 in water is higher than CH4, CH4 will be released as a pure gas and we can extract it. the question is that is it possible to simulate mutual solubilities of CO2, CH4, and salt in the water together with accounting for salt precipitation. Is there any example for this kind of simulation.?

Kind regards
Morteza

**Comment by cpgr** [comment, marked as answer]
Hi @mortezaaesmaeilpour,

I do have this capability that I have used in the past (based on the model of Zirrahi et al, Mutual solubility of CH4, CO2, H2S and their mixtures in brine under subsurface disposal conditions, Fluid Phase Equilibria 324, 2012, pp 80-93), but it isn't publicly available at the moment, and most likely would need some work to make it compatible with the current MOOSE code.

Do you need this right now, or could you wait a month or so for it? I just don't have the time to look into it at the moment.

**Reply by mortezaaesmaeilpour** [reply]
dear Chris

I'm still waiting for your answer.

**Reply by cpgr** [reply]
Sorry, I missed your previous message. I'm certain that the code won't compile without probably several hours of work as it will be in a stale branch somewhere. I could try and dig it out, but I really don't have the time to bring it up to date in the next few weeks.

**Comment by mortezaaesmaeilpour** [comment]
Dear @cpgr 

That's great. Could you please provide it in one or two weeks? I need to convince my supervisor to use the PorousFlow module for the simulation of this problem. Nevertheless, if you have developed another MOOSE-based application, I can use it for now and you have time to integrate it into PorousFlow Module.

Cheers
Morteza

**Reply by GiudGiud** [reply]
@cpgr @mortezaaesmaeilpour can I close this?

**Reply by cpgr** [reply]
I'm going to do this next month @GiudGiud when I get some time - just been crunched by other project deadlines! Feel free to close this if you want.
